crates/omnix-health/src/check/shell.rs

```diff
@@ -0,0 +1,158 @@
+use serde::{Deserialize, Serialize};
+use std::path::{Path, PathBuf};
+
+use crate::traits::{Check, CheckResult, Checkable};
+
+/// Shell types
+///
+#[derive(Debug, Serialize, Deserialize, Clone, Hash, Eq, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum Shell {
+    Zsh,
+    Bash,
+    Undeterminable,
```

Comment:
What does `Undeterminable` actually mean? I don't think should be treated as a shell; it should be an error!